### DISCUSSION TITLE
Show today’s cumulative time grouped by tags, including open ranges

### DISCUSSION THREAD

**Discussion by MoonCactus** [jotaen/klog#365] [Feature ideas & discussions]
Is there a way to show the time spent today grouped by tags ?
I wish the default would also include the one that might still be open, or that queries allow `--now` more systematically (which I understant as `--count-current-task` ;) )

Syntactically it might be something like this:
```
klog tags --since today --now work.klg
```
(the closest I can get is with `klog tags --since 2022-06-30 work.klg`)

Or, arguably less sexy:
```
klog tags --today --now work.klg
```

The reciprocal might be to add `--tags` to the `today` command (since only this one seems to handle `--now`), but it is weird to me:
```
klog today --tags --now work.klg
```

BTW: this tool is terrific, thanks ! :+1:

**Comment by jotaen** [comment]
> Or, arguably less sexy:
> 
> `klog tags --today --now work.klg`

I’m curious, why would you consider this option to be less appealing?

Supporting the `--now` flag for the `today` subcommand would actually seem like the most obvious solution to me. All the other evaluation commands (`total`, `report`, `today`) support the `--now` flag as well, so it would make sense to me to have it for `tags` as well.

**Comment by MoonCactus** [comment]
Well may be. Anyhow, you know better than me anyway :)

BTW I am slightly confused by the syntax. Eg. commands are `total`, `report`, and ... `today`. The first two definitely feel like commands while the last one seems like a filter to me. I tend to think about date as a WHERE clause while tags would be an orthogonal (independent) GROUP BY.

I means there could be a possibily more consistent syntax like:
```
klog command [filters]* 

command = total | report   # among others 
filters = --today | (--since expr) | --diff | --group-by-tags | --now
```
command = what to do, option = grouping and filtering

Eg. as stated in the doc *`klog today` is similar to `klog total`, except that it displays the current day separate from all other days.* Hence I would remember it more easily as an option to `total`, ie. with a syntax like `klog total --today` (reciprocally, it makes a slightly longer command to type).

(my two cents, I can understand I am way too late in the game to change this !)

**Comment by jotaen** [comment]
Thanks for elaborating on your thoughts! The command structure that you suggest is viable as well. I think it’s always a balancing act when to introduce new subcommands, as opposed to extending existing ones with additional flags. I probably lean towards introducing subcommands – that’s mostly for subjective (stylistic) reasons though, because I find that more “tidy”.

Some background info, just for curiosity: the main reason for `klog today` being a dedicated command is that its output structure is a bit different from `klog total`: it’s not just about separating the current day from the other ones, but when you use should total values (e.g. `(7h30m!)`), then `klog today --diff --now` calculates at which point in time your goal will be achieved. Therefore, `klog today` was intended to be a specialised “view” to serve the specific use-case of checking in on the current day.

Btw., the `--today` flag in `klog total --today` is a shortcut for `klog total --date 2022-07-18` (given that today is the 18th of July 2022). The date filters (`--date ...`, `--today`, `--last-week`, etc.) work the same on all subcommands.

**Comment by MoonCactus** [comment]
Thanks for the explanations, I guess it will help me remember the idea !
